hypothesis-python/src/hypothesis/extra/array_api.py

```diff
@@ -735,12 +731,13 @@ def indices(
     )
     check_valid_dims(min_dims, "min_dims")
 
+    ndim = len(shape)
     if max_dims is None:
-        max_dims = min(len(shape), NDIM_MAX)
+        max_dims = min(ndim, NDIM_MAX)
     check_type(int, max_dims, "max_dims")
     assert isinstance(max_dims, int)
     check_argument(
-        max_dims <= len(shape),
+        max_dims <= ndim,
```

Comment:
Let's drop this; calling `len()` is cheap and the existing code is clear.

Reply:
Ah yea, did this for my filter before... I keep on forgetting to revert these otherwise finicky changes heh.